### PR TITLE
Add example of developing multiple functions locally

### LINKED ISSUE #98: How to serve multiple routes on the same Url : Port
Hi, 

I'm trying to serve multiple routes from the same Host / Port combination for local testing. 

Each of these routes would point to a specific target. 
All cloud functions (on GCloud) are sharing the same hostname.  
So I would like to replicate this environment as much as possible 

Is this possible ? I have not seen anything in the documentation to help me perform this setup. 
I would like to serve something like this: 
```
http://localhost:8080/doFunctionA
http://localhost:8080/doFunctionB
http://localhost:8080/doFunctionC
```

Thanks

### PR DESCRIPTION
**Pull request by di**
Fixes #98.

### CODE REVIEW

File: examples/skaffold/README.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# Multiple Cloud Functions, Same Host
```
Comment (grant) [inline examples/skaffold/README.md:1]: The H1 here and example content are not exactly what I would imagine.

Similar to https://github.com/GoogleCloudPlatform/functions-framework-python/issues/98#issuecomment-733872312, I wouldn't expect that the minimum requirement to run multiple functions on the same host involves Kubernetes/`minikube`/`skaffold`. I'd expect a lighter weight setup, like how the FF is very lightweight – maybe require 1 dependency like Docker.

---

That said, can we change this example / PR title to something more aligned to the content, like "Developing Multiple Functions using Minikube and Skaffold"

File: examples/skaffold/README.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+## Introduction
+
+This example shows you how to deploy multiple Cloud Functions to a single host.
```
Comment (grant) [inline examples/skaffold/README.md:5]: This example shows how to develop multiple functions on a Kubernetes cluster with a single host.

- We're not using the Cloud Functions product anymore (FF's can be deployed to many environments).
- This seems like an opinionated way for a multiple function, single host app. Let's label that, as we may want a simpler multiple function, single host app in a different example.

**Review by grant — APPROVED** [review]
Looks like a good skaffold example. 

I'm going to approve but ask that we specifically don't suggest that this solves the multiple functions on a single port issue. This example is an opinionated way to achieve that but requires many dependencies that an existing Functions user is likely not going to want in their application.

**Comment by martindufort** [comment]
Hi all, 

Just want to provide some insights on our in-house solution to enable supporting multiple functions on the same `host:port`.
After investigation, we saw that the `target` function was always called regardless of the url path provided:

```
http://localhost:8080/go
http://localhost:8080/foward
```

Those two requests would both be executed by the `target` function. So we decided to take advantage of this and create a `dispatcher` function in our target file. 

Now we are invoking the functions-framework like this:
``` bash
functions-framework --target dispatcher --port 8081 --debug 
``` 

The `dispatcher` function within our `main.py` file is responsible for introspecting method names and calling the corresponding method if it exists. 

``` python
def dispatcher(request):
    path = request.path
    
    path = path[1:] # Remove prefix / from path
    
    # Introspect all methods for potential execution
    potential_methods = globals().copy()
    potential_methods.update(locals())
    
    # Try to acquire corresponding method associated to the URL path
    method = potential_methods.get(path)
    if not method:
         raise NotImplementedError("URL Path %s not implemented" % path)
    return method(request)
```

File: examples/skaffold/README.md (hunk 1 above)
Comment (di) [inline examples/skaffold/README.md:1]:
```suggestion
# Developing multiple functions on the same host using Minikube and Skaffold
```

File: examples/skaffold/README.md (hunk 2 above)
Comment (di) [inline examples/skaffold/README.md:5]:
```suggestion
This example shows you how to develop multiple Cloud Functions to a single host
using Minikube and Skaffold.
```

**Pull request merged**